src/screens/collection/components/collectible-item.tsx

```diff
@@ -0,0 +1,146 @@
+import { useDispatch } from 'react-redux';
+
+import { OBJKT_CONTRACT } from 'src/apis/objkt/constants';
+import { CollectibleIconSize } from 'src/components/collectible-icon/collectible-icon.props';
+import { Route3TokenStandardEnum } from 'src/enums/route3.enum';
+import { ConfirmationTypeEnum } from 'src/interfaces/confirm-payload/confirmation-type.enum';
+import { FxHashContractInterface, ObjktContractInterface } from 'src/interfaces/marketplaces.interface';
+import { ModalsEnum } from 'src/navigator/enums/modals.enum';
+import { TouchableCollectibleIcon } from 'src/screens/collectibles-home/collectibles-list/touchable-collectible-icon/touchable-collectible-icon';
+import { navigateAction } from 'src/store/root-state.actions';
+import { useSelectedRpcUrlSelector } from 'src/store/settings/settings-selectors';
+import { useSelectedAccountSelector } from 'src/store/wallet/wallet-selectors';
+import { formatSize } from 'src/styles/format-size';
+import { TokenInterface } from 'src/token/interfaces/token.interface';
+import { conditionalStyle } from 'src/utils/conditional-style';
+import { isDefined } from 'src/utils/is-defined';
+import { createTezosToolkit } from 'src/utils/rpc/tezos-toolkit.utils';
+import { getTransferPermissions } from 'src/utils/swap-permissions.util';
+import { mutezToTz } from 'src/utils/tezos.util';
+
+import { useCollectionStyles } from '../collection.styles';
+
+const DEFAULT_OBJKT_STORAGE_LIMIT = 350;
+
+interface Props {
+  item: TokenInterface;
+  collectionContract: string;
+}
+
+export const CollectibleItem: FC<Props> = ({ item, collectionContract }) => {
+  const [offer, setOffer] = useState<ObjktContractInterface | FxHashContractInterface>();
+  const selectedRpc = useSelectedRpcUrlSelector();
+  const selectedAccount = useSelectedAccountSelector();
+  const tezos = createTezosToolkit(selectedRpc);
+  const dispatch = useDispatch();
+
+  const styles = useCollectionStyles();
+
+  useEffect(() => {
+    tezos.contract
+      .at<ObjktContractInterface | FxHashContractInterface>(item.highestOffer?.marketplace_contract ?? OBJKT_CONTRACT)
+      .then(setOffer);
+  }, []);
+
+  const price = isDefined(item.lowestAsk)
+    ? `${mutezToTz(BigNumber(item.lowestAsk), item.decimals)} ${item.symbol}`
+    : '---';
+
+  const highestOffer = isDefined(item.highestOffer)
+    ? `${mutezToTz(BigNumber(item.highestOffer.price), item.decimals)} ${item.symbol}`
+    : 'No offers yet';
+
+  const holders = item?.holders?.map(holder => holder.holder_address) ?? [];
+  const isHolder = holders.includes(selectedAccount.publicKeyHash);
+  const isOffersExisted = isDefined(item.highestOffer);
+
+  const handlePress = async () => {
+    const transferParams = isDefined(offer)
+      ? 'fulfill_offer' in offer?.methods
+        ? [offer.methods.fulfill_offer(item.highestOffer?.bigmap_key ?? 1, item.id).toTransferParams()]
+        : [offer.methods.offer_accept(item.highestOffer?.bigmap_key ?? 1).toTransferParams()]
+      : [];
```

Comment:
Don't use one conditional operator inside another, please. It is hard to read this code